File: packages/rules/test/fixtures/shopify/py/missing-signature-verification.positive.py
from flask import Flask, request

app = Flask(__name__)


@app.route("/webhooks/shopify", methods=["POST"])
def shopify_webhook():
    # BUG: no HMAC verification.
    order = request.get_json()
    print("Shopify order:", order.get("id"))
    return "", 200

File: packages/rules/test/fixtures/shopify/js/missing-signature-verification.positive.ts
import express from "express";

const app = express();
app.use(express.json());

app.post("/webhooks/shopify", async (req, res) => {
  // BUG: no HMAC verification; X-Shopify-Hmac-Sha256 header ignored.
  const order = req.body;
  console.log("Shopify order:", order.id);
  res.status(200).end();
});

app.listen(3000);
